### PR TITLE
Fix several features that depended on specific classes

### LINKED ISSUE #4026: `align-issue-labels` is broken
URL: https://github.com/sindresorhus/refined-github/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc

Screen Shot 1

I'm disabling it via hotfix.

### PR DESCRIPTION
**Pull request by cheap-glitch**
1. LINKED ISSUES: 
	Replaces and closes #4017 
	Fixes #4026 
2. TEST URLS:
 * [`clean-repo-sidebar`](https://github.com/sindresorhus/refined-github)
 * `convert-pr-to-draft-improvements`
 * `parse-backticks`
   * [Commit title above files list](https://github.com/sindresorhus/refined-github/tree/0887b196c66ec3e170da89f6a42ab295fb324fdd)
   * [Commit title above single file](https://github.com/sindresorhus/refined-github/blob/0887b196c66ec3e170da89f6a42ab295fb324fdd/readme.md)
   * ["Pulse" page](https://github.com/sindresorhus/refined-github/pulse)
 * [`first-published-tag-for-merged-pr`](https://github.com/sindresorhus/refined-github/pull/4013)
 * [`align-issue-labels`](https://github.com/sindresorhus/refined-github/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc)

GitHub changed some class names related to links, which broke a few features:
  * `.muted-link` → `.Link--muted` (`clean-repo-sidebar`, `convert-pr-to-draft-improvements`)
  * `.link-gray` → `.Link--secondary` (`parse-backticks`, `easier-pr-sha-copy`)
  * `.link-gray-dark` → `.Link--primary` (`parse-backticks`, `first-published-tag-for-merged-pr`)
  * `.text-gray` → `.color-text-secondary` (`align-issue-labels`, `pr-approvals-count`, `center-reactions-popup`, and probably others)
  * `.text-gray-dark` → `.color-text-primary` (`clean-dashboard`, `parse-backticks`)

### CODE REVIEW

**Comment by yakov116** [comment]
@fregante what do we do about GHE?

**Comment by busches** [comment]
> @fregante what do we do about GHE?

You can see if someone with the new layout can test, otherwise these appear to be "clutter" cleanup features and they won't work. I am personally not as concerned with those breaking and focus more on the new features we add.

**Comment by cheap-glitch** [comment]
Ah yes, forgot about GHE. As I said in my PR most of those we can leave as is without breaking anything (for now), and for the two features that are actually affected I guess we can do something like
```ts
select('.muted-link, .Link--muted');
```

**Comment by fregante** [comment]
Thankfully it’s as easy as using both classes. Only the clean repo sidebar selector is going to be a little longer.

**Comment by fregante** [comment]
It still makes sense to use both classes in our markup, don’t undo that part

File: source/features/clean-repo-sidebar.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ import features from '.';
 
 async function removeReadmeLink(): Promise<void> {
 	// Hide "Readme" link made unnecessary by toggle-files-button #3580
-	(await elementReady('.muted-link[href="#readme"]'))?.parentElement!.remove();
+	(await elementReady('.muted-link[href="#readme"], .Link--muted[href="#readme"]'))?.parentElement!.remove();
```
Comment (yakov116) [inline source/features/clean-repo-sidebar.tsx:11]: They all need comments

Comment (fregante) [reply source/features/clean-repo-sidebar.tsx:11]: I think what’s happening is relatively obvious. Maybe let’s just add a global expiring TODO comment for October or something.

Comment (yakov116) [reply source/features/clean-repo-sidebar.tsx:11]: Fine with me.

Comment (fregante) [reply source/features/clean-repo-sidebar.tsx:11]: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/expiring-todo-comments.md

File: source/features/first-published-tag-for-merged-pr.tsx [hunk 2]
```diff
@@ -21,7 +21,8 @@ const getFirstTag = cache.function(async (commit: string): Promise<string | unde
 });
 
 async function init(): Promise<void> {
-	const mergeCommit = select(`.TimelineItem.js-details-container.Details a[href^="/${getRepo()!.nameWithOwner}/commit/" i] > code.link-gray-dark`)!.textContent!;
+	const timelineItemLinkSelector = `.TimelineItem.js-details-container.Details a[href^="/${getRepo()!.nameWithOwner}/commit/" i]`;
+	const mergeCommit = select(`${timelineItemLinkSelector} > code.Link--primary, ${timelineItemLinkSelector} > code.link-gray-dark`)!.textContent!;
```
Comment (yakov116) [inline source/features/first-published-tag-for-merged-pr.tsx:25]: look at #4017 just remove the `.link-gray-dark` I added it when I made the feature for no reason. (Unless you see why its needed)

**Comment by yakov116** [comment]
if only we could use `:has()` 😞

File: source/features/clean-repo-sidebar.tsx (hunk 1 above)
Comment (yakov116) [reply source/features/clean-repo-sidebar.tsx:11]: Maybe by the fetureRenamed?

**Review by fregante — APPROVED** [review]
Unverified but the code looks good so far. Maybe we can wait a couple more days to see if there are any new classes that need to be changed.

**Comment by cheap-glitch** [comment]
Another class seems to have changed: `.text-gray` became `.color-secondary-text` in some places (but not everywhere).

I've already fixed `pr-approvals-count` and `align-issue-labels` that depended on it (messed up my commit message, it should say "text-gray" instead of "link-gray"), but there are other features that may be affected and that I'm not very familiar with.

**Comment by fregante** [comment]
Are you saying this PR fixes #4026?

**Comment by cheap-glitch** [comment]
> Are you saying this PR fixes #4026?

Yep

**Comment by fregante** [comment]
> if only we could use `:has()` 😞

Hey, we can! 🎉 

Screen Shot 2

We just need to update the minimum versions in the manifest

**Comment by fregante** [comment]
Possibly unrelated to this PR, but might also need class changes:

Are you also seeing these issues? (`align-issue-labels` is disabled)

- "Approved" isn't colored
- We have a feature that adds duplicate checkboxes

Screen Shot 4

**Comment by cheap-glitch** [comment]
> Are you also seeing these issues?
>* "Approved" isn't colored 
>* We have a feature that adds duplicate checkboxes

First one (`pr-approvals-count`) is already fixed. I'm not seeing the other.

**Comment by cheap-glitch** [comment]
@fregante I'm seeing `.link-gray` in `easier-pr-sha-copy.css` but I can't figure out what it's supposed to target. Is this feature still relevant? Not seeing any SHAs on the PRs list page.

**Comment by fregante** [comment]
It’s worth checking out its blame to understand it

File: source/features/easier-pr-sha-copy.css [hunk 3]
```diff
@@ -1,9 +1,4 @@
-/*
-This feature has to work on:
-https://github.com/sindresorhus/refined-github/pulls
-*/
-
-/* Add margin to commit SHA to reduce likelihood of selection overflow */
-.repository-content .js-commit .text-right code .link-gray {
+/* Add margin to commit SHA to reduce likelihood of selection overflow #2508 */
+.repository-content .js-commit .text-right code .Link--secondary {
```
Comment (fregante) [inline source/features/easier-pr-sha-copy.css:2]: `link-gray` for  GHE too, right?

**Comment by fregante** [comment]
Let me know when you think you're done on this PR, I'd like to merge this soon (ideally together with the `batch-conversation-open` feature)

**Comment by fregante** [comment]
🙌 

Merging for the time being because I keep seeing bugs and I don't if this already fixed them.

**Comment by fregante** [comment]
There are definitely instances of `text-gray` and others, an additional global search/replace should fix more issues, including the duplicate checkboxes.

**Comment by cheap-glitch** [comment]
> There are definitely more instances of `text-gray` and others

I'm aware, but I didn't want to blindly search & replace every instance seeing as there's still some `.text-gray` classes in the DOM (e.g. pinned issues), and that this affects features I'm not familiar with.

**Comment by fregante** [comment]
No worries, thanks for the detailed PR, it will be useful when blaming these changes.

**Pull request merged**